moveit_py/src/moveit/moveit_core/kinematic_constraints/utils.cpp

```diff
@@ -0,0 +1,159 @@
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Peter David Fagan */
+
+#include "utils.h"
+#include <rclcpp/rclcpp.hpp>
+#include <moveit_py/moveit_py_utils/ros_msg_typecasters.h>
+#include <moveit/kinematic_constraints/utils.h>
+
+namespace moveit_py
+{
+namespace bind_kinematic_constraints
+{
+moveit_msgs::msg::Constraints construct_link_constraint(const std::string& link_name, const std::string& source_frame,
+                                                        std::optional<std::vector<double>> cartesian_position,
+                                                        std::optional<double> cartesian_position_tolerance,
+                                                        std::optional<std::vector<double>> orientation,
+                                                        std::optional<double> orientation_tolerance)
+{
+  // check that link cartesian and/or orientation constraints are specified
+  if (!cartesian_position && !orientation)
+  {
+    throw std::invalid_argument("No link cartesian or orientation constraints specified");
+  }
+
+  moveit_msgs::msg::Constraints constraints_cpp;
+
+  // merge constraints if necessary
+  if (cartesian_position && orientation)
+  {
+    // define point stamped message
+    geometry_msgs::msg::PointStamped point;
+    point.header.frame_id = source_frame;
+    point.point.x = cartesian_position.value()[0];
+    point.point.y = cartesian_position.value()[1];
+    point.point.z = cartesian_position.value()[2];
+
+    moveit_msgs::msg::Constraints position_constraints =
+        kinematic_constraints::constructGoalConstraints(link_name, point, cartesian_position_tolerance.value());
+
+    // define quaternion message
+    geometry_msgs::msg::QuaternionStamped quaternion;
+    quaternion.header.frame_id = source_frame;
+    quaternion.quaternion.x = orientation.value()[0];
+    quaternion.quaternion.y = orientation.value()[1];
+    quaternion.quaternion.z = orientation.value()[2];
+    quaternion.quaternion.w = orientation.value()[3];
+
+    moveit_msgs::msg::Constraints orientation_constraints =
+        kinematic_constraints::constructGoalConstraints(link_name, quaternion, orientation_tolerance.value());
+
+    constraints_cpp = kinematic_constraints::mergeConstraints(position_constraints, orientation_constraints);
+  }
+
+  // generate cartesian constraint
+  else if (cartesian_position)
+  {
+    // define point stamped message
+    geometry_msgs::msg::PointStamped point;
+    point.header.frame_id = source_frame;
+    point.point.x = cartesian_position.value()[0];
+    point.point.y = cartesian_position.value()[1];
+    point.point.z = cartesian_position.value()[2];
+
+    // instantiate logger
+    auto logger = rclcpp::get_logger("moveit_py");
+    // check point with logger
+    RCLCPP_INFO(rclcpp::get_logger("moveit_py"), "Point: %f, %f, %f", point.point.x, point.point.y, point.point.z);
```

Comment:
Is this something we want to merge, or was it a debug message? I'd say either remove it or use `RCLCPP_DEBUG`